### ISSUE TITLE
Sensitive data stored in the device in clear

### ISSUE THREAD

**Issue by javixeneize** [timb-machine/nhsx-contact-tracing-app#35]
This is related to issue #21. Other sensitive data is stored in the device as the user report including the symptoms, or a file called persistedInstallation.xxxx.json, which contains an auth token and a refresh token

This issue has also been reported here https://www.stateofit.com/UKContactTracing/#Data-stored-unencrypted

**Comment by marksteward** [comment]
This can potentially be mitigated by encrypting the contact entries to the server's public key, but I'm not sure it's possible to encrypt the tokens. Generally, if someone has root over a device they can do anything the app can do.

**Comment by javixeneize** [comment]
If it stores just an identifier or uses a private key generated in the secure enclave /TEE in the device to encrypt the data, it should solve the problem.

**Comment by marksteward** [comment]
If it uses a private key, someone with root on the device can decrypt it.

**Comment by javixeneize** [comment]
No, if the keypair is in the secure enclave or TEE, it can't be decrypted

**Comment by marksteward** [comment]
I mean if it uses a private keypair stored in TEE to encrypt, the same app can decrypt it, because it has access to the private key.

**Comment by javixeneize** [comment]
Yeah, but it will reduce a lot the scope of the threat model in my opinion. I think that’s the best that can be done with rooted devices
